pennylane/ops/functions/equal.py

```diff
@@ -482,3 +485,24 @@ def _equal_shadow_measurements(op1: ShadowExpvalMP, op2: ShadowExpvalMP, **_):
 @_equal.register
 def _equal_counts(op1: CountsMP, op2: CountsMP, **kwargs):
     return _equal_measurements(op1, op2, **kwargs) and op1.all_outcomes == op2.all_outcomes
+
+
+@_equal.register
+def _equal_basis_rotation(
+    op1: qml.BasisRotation,
+    op2: qml.BasisRotation,
+    check_interface=True,
+    check_trainability=True,
+    rtol=1e-5,
+    atol=1e-9,
+):
+    if not np.allclose(
+        op1._hyperparameters["unitary_matrix"],  # pylint: disable=protected-access
+        op2._hyperparameters["unitary_matrix"],  # pylint: disable=protected-access
```

Comment:
```suggestion
        op1.hyperparameters["unitary_matrix"],
        op2.hyperparameters["unitary_matrix"], 
```

`hyperparameters` is the public property corresponding to the private `hyperparameters`.